### ISSUE TITLE
Long durations in the ban command result in the user being able to join after being banned

### ISSUE THREAD

**Issue by KarmaWSYD** [SmartlyDressedGames/Unturned-3.x-Community#4806]
Long durations (e.g. 2647483647, tested by incrementing 2^31-1 with 100 million until an issue occurred) in the ban command result in the user being able to rejoin the server after being banned. The ban command still goes through and shows up properly on both the server and the client, but the client can join the server immediately after.

While the current maximum duration (~80 years) isn't necessarily an issue the ban appearing to work properly when in reality it doesn't is a problem.

The (vanilla) server this was tested on is running on ubuntu with pterodactyl.

**Comment by SDGNelson** [comment]
Unfortunately, it's one of a few features from very early in development that uses 32-bit unix time. The ban system would certainly benefit from a tidying up. :x

**Comment by Jdance-Media** [comment]
I obviously don't know much about the blacklist system, but shouldn't it be relatively easy to change the integral type nowadays? 32-bit support was dropped in 2022 I believe. The class which uses the 32-bit type can be changed without problem because you read from the Block/River and then make the class from that, so you can convert prior.

**Comment by SDGNelson** [comment]
@Jdance-Media that is fair. Ideally, I'd like to address a few things at once like using a `System.DateTime` for time here, and displaying the amount of time remaining on the ban in a nicer way. (as opposed to the current number of seconds without units)

**Comment by Jdance-Media** [comment]
Fair enough, implementing the usage of DateTime would be very nice. Should in theory be able to just store the ticks.
